src/trading/NewOrders.sol

```diff
@@ -0,0 +1,338 @@
+            _fill = _sharesFilled + (_tokensFilled.fxpDiv(_market.getMaxDisplayPrice() - orders[_orderId].fxpPrice));
+        }
+        require(_fill <= orders[_orderId].fxpAmount);
+        orders[_orderId].fxpAmount -= _fill;
+        orders[_orderId].fxpMoneyEscrowed -= _tokensFilled;
+        orders[_orderId].fxpSharesEscrowed -= _sharesFilled;
+        if (orders[_orderId].fxpAmount == 0) {
+            require(orders[_orderId].fxpMoneyEscrowed == 0);
+            require(orders[_orderId].fxpSharesEscrowed == 0);
+            removeOrderFromList(_orderId, _orderType, _market, _orderOutcome);
+            orders[_orderId].fxpPrice = 0;
+            orders[_orderId].owner = 0;
+            orders[_orderId].gasPrice = 0;
+            orders[_orderId].betterOrderId = 0;
+            orders[_orderId].worseOrderId = 0;
+        }
+        return true;
+    }
+
+    function takeOrderLog(IMarket _market, uint8 _orderOutcome, Trading.TradeTypes _orderType, bytes20 _orderId, address _taker, uint256 _makerSharesFilled, uint256 _makerTokensFilled, uint256 _takerSharesFilled, uint256 _takerTokensFilled, uint256 _tradeGroupId) public constant returns (bool) { 
+        uint256 _price = orders[_orderId].fxpPrice;
+        address _maker = orders[_orderId].owner;
+        TakeOrder(_market, _orderOutcome, _orderType, _orderId, _price, _maker, _taker, _makerSharesFilled, _makerTokensFilled, _takerSharesFilled, _takerTokensFilled, _tradeGroupId);
+        return true;
+    }
+
+    function completeSetsLog(address _sender, IMarket _market, Trading.TradeTypes _type, uint256 _fxpAmount, uint256 _numOutcomes, uint256 _marketCreatorFee, uint256 _reportingFee) public constant returns (bool) { 
+        CompleteSets(_sender, _market, _type, _fxpAmount, _numOutcomes, _marketCreatorFee, _reportingFee);
+        return true;
+    }
+
+    function cancelOrderLog(IMarket _market, address _sender, uint256 _fxpPrice, uint256 _fxpAmount, bytes20 _orderId, uint8 _outcome, Trading.TradeTypes _type, uint256 _fxpMoneyEscrowed, uint256 _fxpSharesEscrowed) public constant returns (bool) { 
+        CancelOrder(_market, _sender, _fxpPrice, _fxpAmount, _orderId, _outcome, _type, _fxpMoneyEscrowed, _fxpSharesEscrowed);
+        return true;
+    }
+
+    function modifyMarketVolume(IMarket _market, uint256 _fxpAmount) internal returns (bool) { 
+        marketOrderData[_market].volume += _fxpAmount;
+        _market.getBranch().getTopics().updatePopularity(_market.getTopic(), _fxpAmount);
+        return true;
+    }
+
+    function setPrice(IMarket _market, uint8 _outcome, uint256 _fxpPrice) internal returns (bool) { 
+        marketOrderData[_market].prices[_outcome] = _fxpPrice;
+        return true;
+    }
+
+    function removeOrderFromList(bytes20 _orderId, Trading.TradeTypes _type, IMarket _market, uint8 _outcome) private returns (bool) { 
```

Comment:
Nit: A lot of lines in this file have an extraneous space at the end, before the newline.